src/main/kotlin/com/kruthers/gamemode4core/events/DimensionEvents.kt

```diff
@@ -22,6 +23,17 @@ class DimensionEvents(val plugin: Gamemode4Core): Listener {
             if (Gamemode4Core.watchingPlayers.values.contains(player.uniqueId)) {
                 Gamemode4Core.watchingPlayers.forEach { watcher, target ->
                     if (target == player.uniqueId) {
+                        watcher.spectatorTarget?.let {
+                            watcher.spectatorTarget = null
+                            Bukkit.getScheduler().runTaskLater(plugin, Runnable {
```

Comment:
instead of delaying 1 tick after you can get `event.to` and tp to there immediately. if you adjust the listener and check its not cancled. Then you only need 1 delay to make them a specator them again.